### ISSUE TITLE
Feature request: intermediate model saving and early stopping

### ISSUE THREAD

**Issue by sudarshan85** [fastai/fastai#457]
Hi,

I would like to have a feature where during training, the last n models with lowest validation error is held (either stored or in RAM) and after training is finished have an ability to plot the training loss and validation loss in a multiplot of all the held models. This is very similar to early stopping.

I'm not sure this feature is available in the library. If not, I'd like to implement it. I would like some guidance as to where to start and some pointers on how to do it.

Thanks.

**Comment by sean-adler** [comment]
[`SaveBestModel`](https://github.com/fastai/fastai/blob/master/fastai/sgdr.py#L292) almost does what you want, so you'd need to extend it to save the `n` best models and then plot them.
